cloudinit/distros/__init__.py

```diff
@@ -994,23 +993,17 @@ def is_virtual(self) -> bool:
             # For non systemd systems the method should be
             # implemented in the distro class.
             LOG.debug("is_virtual should be implemented on distro class")
-            return True
+            return None
 
         try:
-            proc = subprocess.run(
-                "systemd-detect-virt",
-                stdin=None,
-                shell=True,
-                capture_output=True,
-                check=False,
-                text=True,
+            (out, _) = subp.subp(
+                ["systemd-detect-virt"], shell=True, capture=True, rcs=[0, 1]
```

Comment:
why do we need the shell here?

Reply:
The path to the binary differs from distribution to distribution, my understanding is that shell loads the PATH environment variable correctly. Am I wrong?